### PR TITLE
Add recipe for squadgen

### PR DESCRIPTION
**Pull request by xylar**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by xylar** [comment]
Note: this library includes the source for the legacy NetCDF C++ library, but it doesn't create a library of its own, just an executable.  The license file has been included.

**Comment by xylar** [comment]
@conda-forge/help-c-cpp, ready for review!

**Comment by xylar** [comment]
@ocefpaf, another one that you might be willing to have a look at?  Also, if you understand what I've done that is causing `libpng` to be linked statically rather than dynamically, let me know.  I'm not super expert in CMake yet...

File: recipes/squadgen/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  commands:
+    - >
+      SQuadGen --refine_file test/grid.png --refine_level 5
+      --refine_type LOWCONN --resolution 30 --smooth_type SPRING
+      --smooth_dist 3 --smooth_iter 20 --output grid.g
+    - test -f "grid.g"
+
+about:
+  home: https://github.com/ClimateGlobalChange/squadgen
+  license: GPL-2.0-or-later and CUSTOM
```
Comment (xhochy) [inline recipes/squadgen/meta.yaml:45]:
```suggestion
  license: GPL-2.0-or-later and NetCDF
```
Isn't this the right one?

**Review by xhochy — COMMENTED** [review]
Maybe try BUILD_SHARED_LIBS=1

File: recipes/squadgen/meta.yaml (hunk 1 above)
Comment (xylar) [reply recipes/squadgen/meta.yaml:45]: Thanks you @xhochy!  These are very helpful suggestions!

**Comment by xylar** [comment]
@xhochy, thanks again for your help.  

I finally solved the mystery.  `squadgen` was including `png.h` but then not actually using it for anything (so I have patched to remove the unneeded include).  Instead, it builds [LodePNG](https://lodev.org/lodepng/) internally and uses that.  I have added the license for this as well.

**Comment by xylar** [comment]
@xhochy, the tests are now passing and there is no longer any overdependency.  Would you be able to give this another look?  Thank you!

**Comment by ocefpaf** [comment]
Thanks @xhochy for the review and @xylar for the package!

**Comment by xylar** [comment]
Thanks so much @ocefpaf and @xhochy!

**Pull request merged**